compiler/rustc_symbol_mangling/src/typeid.rs

```diff
@@ -0,0 +1,25 @@
+// For more information about type metadata and type metadata identifiers for cross-language LLVM
+// CFI support, see the design document at: https://rcvalle.com/docs/rust-cfi-design-doc.pdf.
```

Comment:
Would it be possible to place this document either in-repository or some other location that's more permanent than a personal website? It would be a shame if we ended up needing to look things up a dozen years down the line and found a 404.

Reply:
Sure! Makes sense. What do you think about submitting it as a Markdown doc when it's finished to somewhere in src/doc? Where would be the best place for it there?

Reply:
I changed the comments to point to the tracking issue instead while we figure out a place for it (and then we just update the tracking issue).